src/emulator/cpu.cpp

```diff
@@ -18,4 +18,28 @@ void CPU::AdvanceProgramCounter()
 void CPU::Reset()
 {
     registers_.pc = memory_->ReadWord(kReset);
+}
+
+//Handle nom-maskable interrupts
+void CPU::Interrupt()
+{
+    // TODO - This is just copied from the opcode BRK implementation.  Can we just reuse or does anything change?  Open for further investiation.
+    SetStatusRegisterFlag(kBreakFlag);
```

Comment:
It looks like the break flag is not set on NMI, see here: https://wiki.nesdev.org/w/index.php/Status_flags#The_B_flag